### PR TITLE
feat(connector): [Authorizedotnet] Add Wallet support

### PR DESCRIPTION
**Pull request by srujanchikke**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Wallet Support[Paypal, Googlepay and Applepay] for Authorizedotnet

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

**Note** : It takes one day for this connector to update status from Processing to Success.
Google pay 
Screenshot 2023-05-21 at 11 10 00 PM

Paypal Redirection 
Screenshot 2023-05-21 at 11 08 52 PM

Screenshot 2023-05-21 at 11 08 39 PM

Screenshot 2023-05-21 at 11 09 10 PM

tested unit tests to ensure existing flows
Screenshot 2023-05-21 at 11 15 58 PM

tested airwallex as there is core change which may alter airwallex 3ds payment 

Screenshot 2023-05-29 at 12 28 37 PM
Screenshot 2023-05-29 at 12 29 05 PM

**Note**  : Apple pay testing is pending

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/authorizedotnet/transformers.rs [hunk 1]
```diff
@@ -53,24 +61,43 @@ struct CreditCardDetails {
...
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct PayPalDetails {
+    pub success_url: Option<String>,
+    pub cancel_url: Option<String>,
+}
+
+#[derive(Serialize, Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+
```
Comment (SanchithHegde) [inline crates/router/src/connector/authorizedotnet/transformers.rs:89]: Please remove this newline.

File: crates/router/src/connector/authorizedotnet/transformers.rs [hunk 2]
```diff
@@ -338,50 +374,68 @@ impl From<AuthorizedotnetPaymentStatus> for enums::AttemptStatus {
...
+    pub account_number: Option<String>,
+    pub errors: Option<Vec<ErrorMessage>>,
 }
 
-#[derive(Debug, Clone, Deserialize, PartialEq)]
+#[derive(Debug, Clone, Deserialize)]
+pub struct SecureAcceptance {
+    #[serde(rename = "SecureAcceptanceUrl")]
+    secure_acceptance_url: Option<url::Url>,
+}
```
Comment (SanchithHegde) [inline crates/router/src/connector/authorizedotnet/transformers.rs:436]: Use `#[serde(rename_all = "PascalCase")]` instead.

**Review by SanchithHegde — APPROVED** [review]

File: crates/router/src/connector/authorizedotnet/transformers.rs [hunk 3]
```diff
@@ -53,24 +61,42 @@ struct CreditCardDetails {
...
-#[derive(Serialize, Deserialize, PartialEq, Debug)]
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
 enum PaymentDetails {
-    #[serde(rename = "creditCard")]
     CreditCard(CreditCardDetails),
-    #[serde(rename = "bankAccount")]
     BankAccount(BankAccountDetails),
-    Wallet,
-    Klarna,
```
Comment (jagan-jaya) [inline crates/router/src/connector/authorizedotnet/transformers.rs:69]: why are other PaymentDetails removed? aren't they supported now?

File: crates/router/src/connector/authorizedotnet/transformers.rs [hunk 4]
```diff
@@ -53,24 +61,42 @@ struct CreditCardDetails {
...
-    Paypal,
-    #[serde(rename = "bankRedirect")]
+    OpaqueData(WalletDetails),
+    PayPal(PayPalDetails),
     BankRedirect,
-    BankTransfer,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
```
Comment (jagan-jaya) [inline crates/router/src/connector/authorizedotnet/transformers.rs:80]: need both Serialize and Deserialize

File: crates/router/src/connector/authorizedotnet/transformers.rs [hunk 5]
```diff
@@ -53,24 +61,42 @@ struct CreditCardDetails {
...
+#[serde(rename_all = "camelCase")]
+pub struct PayPalDetails {
+    pub success_url: Option<String>,
+    pub cancel_url: Option<String>,
+}
+
+#[derive(Serialize, Debug, Deserialize)]
```
Comment (jagan-jaya) [inline crates/router/src/connector/authorizedotnet/transformers.rs:87]: need both Serialize and Deserialize

File: crates/router/src/connector/authorizedotnet/transformers.rs [hunk 6]
```diff
@@ -53,24 +61,42 @@ struct CreditCardDetails {
...
+#[serde(rename_all = "camelCase")]
+pub struct WalletDetails {
+    pub data_descriptor: WalletMethod,
+    pub data_value: String,
+}
+
+#[derive(Serialize, Debug, Deserialize)]
```
Comment (jagan-jaya) [inline crates/router/src/connector/authorizedotnet/transformers.rs:94]: need both Serialize and Deserialize

**Review by jagan-jaya — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/authorizedotnet/transformers.rs (hunk 3 above)
Comment (srujanchikke) [reply crates/router/src/connector/authorizedotnet/transformers.rs:69]: Other payment methods are not implemented for this connector. They don't support this payment methods

File: crates/router/src/connector/authorizedotnet/transformers.rs (hunk 5 above)
Comment (srujanchikke) [reply crates/router/src/connector/authorizedotnet/transformers.rs:87]: Parent struct uses both deserialize and serialize .

File: crates/router/src/core/payments.rs [hunk 7]
```diff
@@ -418,7 +418,7 @@ impl PaymentRedirectFlow for PaymentRedirectCompleteAuthorize {
             // If the status is terminal status, then redirect to merchant return url to provide status
             api_models::enums::IntentStatus::Succeeded
             | api_models::enums::IntentStatus::Failed
-            | api_models::enums::IntentStatus::Cancelled | api_models::enums::IntentStatus::RequiresCapture=> helpers::get_handle_response_url(
+            | api_models::enums::IntentStatus::Cancelled | api_models::enums::IntentStatus::RequiresCapture| api_models::enums::IntentStatus::Processing=> helpers::get_handle_response_url(
```
Comment (ArjunKarthik) [inline crates/router/src/core/payments.rs:421]: Please get an approval from core team for this

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]

File: crates/router/src/build.rs [hunk 8]
```diff
@@ -2,7 +2,7 @@ fn main() {
     // Set thread stack size to 4 MiB for debug builds
     // Reference: https://doc.rust-lang.org/std/thread/#stack-size
     #[cfg(debug_assertions)]
-    println!("cargo:rustc-env=RUST_MIN_STACK=4194304"); // 4 * 1024 * 1024 = 4 MiB
+    println!("cargo:rustc-env=RUST_MIN_STACK=6000000"); // 4 * 1024 * 1024 = 4 MiB
```
[the remark this answers is not in the document]
Comment (srujanchikke) [reply crates/router/src/build.rs]: i reverted this change.

File: crates/router/src/core/payments.rs (hunk 7 above)
Comment (Narayanbhat166) [reply crates/router/src/core/payments.rs:421]: Yes, this is fine.

**Review by SanchithHegde — DISMISSED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by jagan-jaya — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Pull request merged**